pythia/cleaning/midnight_rotation.py

```diff
@@ -0,0 +1,111 @@
+import datetime
+
+import astropy.units as u
+from sunpy.physics.differential_rotation import diff_rot
+
+__all__ = ['MidnightRotation']
+
+
+class MidnightRotation:
+    """
+    A class to rotate the coordinates of an observation
+    to its coordinates at the nearest midnight.
+    """
+    def get_nearest_midnight(self, obsdate: str, fmt='%Y-%m-%d %H:%M:%S'):
+        """
+        For a given observation time and date, returns the datetime of the nearest midnight.
+
+        Parameters
+        ----------
+        obsdate : str
+            The observation time and date.
```

Comment:
why not get a `datetime` object instead?

Reply:
It's easier to pass strings as we have strings in the csv.